core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -332,7 +332,8 @@ protected void validateDataFilesExist(TableMetadata base, Long startingSnapshotI
 
     ManifestGroup matchingDeletesGroup = new ManifestGroup(ops.io(), manifests, ImmutableList.of())
         .filterManifestEntries(entry -> entry.status() != ManifestEntry.Status.ADDED &&
-            newSnapshots.contains(entry.snapshotId()) && requiredDataFiles.contains(entry.file().path()))
+            newSnapshots.contains(entry.snapshotId()) &&
+            requiredDataFiles.stream().anyMatch(entry.file().path().toString()::contains))
```

Comment:
This was correct because the set passed in is actually a `CharSequenceSet`: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/BaseRowDelta.java#L29

To fix the error-prone warning and ensure that future uses are also `CharSequenceSet`, I think that a better fix is to change `Set<CharSequence>` to `CharSequenceSet` in the method arguments.

Reply:
I think we might want to adjust all the places where `Set<CharSequence>` is being used just to make sure. I went ahead and adjusted all production code.